sklearn/compose/_column_transformer.py

```diff
@@ -350,7 +350,7 @@ def set_output(self, *, transform=None):
             - `"default"`: Default output format of a transformer
             - `"pandas"`: DataFrame output
             - `"polars"`: Polars output
-            - `None`: Transform configuration is unchanged
+            - `None`: Transform configuration is unchanged, global configuration is used
```

Comment:
Actually it is not necessarily using the global configuration, you could imagine something like this:

```py
estimator.set_output(transform='pandas')
estimator.set_output(transform=None)
```

The global configuration is `transform='default'` but the estimator is using `transform='pandas'`

More a meta comment: I get the fact that it may not be very clear from the docstring what `transform=None` but I don't think there is an easy way to improve the situation in the docstring. Full disclosure: I am not sure about the use case for `transform=None` actually :thinking:.

Reply:
Oh, I was thinking that we cannot actively set `transform=None`, because this is what the docstring says: `transform : {"default", "pandas", "polars"}, default=None`. 

Also, there is
```
    supported_outputs = ADAPTERS_MANAGER.supported_outputs
    if dense_config not in supported_outputs:
        raise ValueError(
            f"output config must be in {sorted(supported_outputs)}, got {dense_config}"
        )
```
to prevent us from setting `transform=None`.

It can only stay `None` if it is untouched, and then the global configuration is used within the `_get_output_config()` function.

I'm happy to revert this if it's wrong, but I will keep my suggestions for now to facilitate discussion.

Reply:
> transform : {"default", "pandas", "polars"}, default=None

I guess the docstring is slightly misleading then. It seems like you can use `transform=None`:

```py
from sklearn.datasets import load_iris
from sklearn.preprocessing import StandardScaler

X, y = load_iris(as_frame=True, return_X_y=True)

scaler = StandardScaler()
scaler.set_output(transform="pandas")
scaler.set_output(transform=None)

scaler.fit(X)
# this outputs a pandas DataFrame so transform=None did not change transform="pandas"
X_scaled = scaler.transform(X)
```

Reply:
`None` is used as a sentinel. If we had `predict` support in `set_output`, then the default `transform=None` will leave transform configuration alone:

```python
scaler.set_output(predict="pandas") 
```

Reply:
Your example got me really confused @lesteve , because I had tried a similar experiment with `FunctionTransformer`, that had led me to think that we cannot pass `transform=None`:

```python
from sklearn.datasets import load_iris
from sklearn.preprocessing import FunctionTransformer

#set_config(transform_output="pandas")  # same as using set_output before transform=None 
X, y = load_iris(as_frame=True, return_X_y=True)

functrans = FunctionTransformer()
functrans.set_output(transform="pandas")
functrans.set_output(transform=None)

functrans.fit(X)
# this raises a ValueError:
functrans.transform(X)
```

`ValueError: output config must be in ['default', 'pandas', 'polars'], got None`

It seems that `FunctionTransformer().set_output()` and `_SetOutputMixin.set_output() `(from `StandardScaler` ) are handeling this differently. The latter has an early [`if transform is None: return self`](https://github.com/scikit-learn/scikit-learn/blob/64ab789905077ba8990522688c11177442e5e91f/sklearn/utils/_set_output.py#L411), while the `FunctionTransformer` can pass self._sklearn_output_config = {transform: None} to `_get_output_config()`, which would then raise the above error message.

I am not sure what to make of this, though...
Is this wanted/expected or should we address this?

I think I should revert the documentation change, since `transform=None` is sometimes possible to pass and then the previous setting is used, not necessarily the global configuration, right?

Reply:
OK well, I think we should fix the bug first and leave the docstring modification for another PR when we understand a bit more the `.set_output` thing.

Reply:
Yes, I agree. I will revert the documentation changes then.